### ISSUE TITLE
Server-side redirect

### ISSUE THREAD

**Issue by IlyaSemenov** [frandiox/vite-ssr#55]
`vite-ssr` currently lacks any reasonable option to perform a server-side redirect.

Even though `response` object is passed in context, one can't use it like:

```ts
export default viteSSR(App, { routes }, ({ response }) => {
  // Simplified version
  // In real code, that could be injected/provided as a consumable hook for Vue pages
  response.setHeader("location", "/foo")
  response.statusCode = 302
  response.end()
})
```

because vite-ssr will always try to end it itself:

https://github.com/frandiox/vite-ssr/blob/d758af9fc102700aa666ac92d5efa6fa55da67a7/src/dev/server.ts#L103-L107

Suggestions:

* at the very least, wrap the 3 last 3 lines like:

```ts
  if (!response.writableEnded) {
     const html = buildHtmlDocument(template, htmlParts)
     response.setHeader('Content-Type', 'text/html')
     response.end(html)
  }
```

* or rather, allow to inject `status`, `statusText`, `headers` and `body` not only from `options.getRenderContext` but also from `viteSSR`.

**Comment by IlyaSemenov** [comment]
Ideally, both `vite-ssr/vue` and `vite-ssr/react` integration should simply provide `redirect` functions which will work both in SSR and on client side, similar to how Nuxt works.

**Comment by frandiox** [comment]
@IlyaSemenov Can you have a look at this? https://github.com/frandiox/vite-ssr/pull/59
It's only implemented for Vue but I'd like to get some feedback to see if that's enough for general use-cases.

The `redirect` function can be used from the main hook or from any component. In components, it can be injected with a new `useContext()` hook. I think the same should be possible in React.

**Comment by IlyaSemenov** [comment]
@frandiox Since you came up with a relatively deep refactoring I believe it's worth it to take one step more and 'generalise' it to to support arbitrary status and/or text (non-Vue-rendered) responses. One particular use case I have in mind is rendering HTML with Vue but returning it with status 404 in SSR requests against URLs like `/show-item/:id` where item with given id doesn't exist.

**Comment by frandiox** [comment]
Implemented in #59 🎉 
Thanks for all the help!
